src/pegen/metagrammar.gram

```diff
@@ -112,19 +110,22 @@ atom[Plain]:
     | NAME {NameLeaf(name.string) }
     | STRING {StringLeaf(string.string)}
 
-# Mini-grammar for the actions
+# Mini-grammar for the actions and annotations
 
 action[str]: "{" ~ target_atoms "}" { target_atoms }
+annotation[str]: "[" ~ target_atoms "]" { target_atoms }
 
 target_atoms[str]:
     | target_atom target_atoms { target_atom + " " + target_atoms }
     | target_atom { target_atom }
 
 target_atom[str]:
-    | "{" ~ target_atoms "}" { "{" + target_atoms + "}" }
+    | "{" ~ atoms=target_atoms? "}" { "{" + (atoms or "") + "}" }
+    | "[" ~ atoms=target_atoms? "]" { "[" + (atoms or "") + "]" }
+    | NAME "*" { name.string + "*" }
```

Comment:
This is not actually needed, though makes it look pretty (`something * ` => `something*`)